wal/wal.go

```diff
@@ -532,21 +540,28 @@ func (w *WAL) saveState(s *raftpb.HardState) error {
 }
 
 func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
-	w.mu.Lock()
```

Comment:
I'd rather avoid complicating the code with sync.Pool unless there's a meaningful performance impact when running the etcd server.